File: f.py
import random
import matplotlib.pyplot as plt
import time

def partition(a,low,high):
    pivot=a[low]
    i=low
    j=high
    while(i<j):
        while(a[i]<pivot):
            i+=1
        while(a[j]>pivot):
            j-=1
        if(i<j):
            a[i],a[j]=a[j],a[i]
    a[j],pivot=pivot,a[j]
    return j
def qs(a,low,high):
    if(low<high):
        m=partition(a,low,high)
        qs(a,low,m-1)
        qs(a,m+1,high)

retx=[]
ret=[]
for i in range(0,21):
    p=[]
    for l in range(0,i):
        p.append(random.randint(1,1000000))
    print(p)
    t1=time.perf_counter()
    qs(p,0,i-1)
    t2=time.perf_counter()
    print(p)
    retx.append(i)
    ret.append(t2-t1)
print(retx)
print(ret)

plt.plot(retx,ret)
plt.show()

File: f.c
#include<stdio.h>
#include<time.h>
#include<stdlib.h>
double tt,mm;
int partition(int a[],int low,int high){
    int pivot=a[low];
    int i=low;
    int j=high;
    while(i<j){
        while(a[i]<pivot){
            i++;
        }
        while(a[j]>pivot){
            j--;
        }
        if(i<j){
            int temp=a[i];
            a[i]=a[j];
            a[j]=temp;
        }
    }
    int temp=pivot;
    pivot=a[j];
    a[j]=temp;
    return j;
}
void qs(int a[],int low,int high){
    if(low<high){
        int c=partition(a,low,high);
        qs(a,low,c-1);
        qs(a,c+1,high);
    }
}
int main(){
    int n;
    printf("enter the size\n");
    scanf("%d",&n);
    int a[n];
    for(int i=0;i<n;i++){
        a[i]=(rand()%100000000)+1;
    }
    for(int i=0;i<n;i++){
        printf(" %d ",a[i]);
    }
    printf("\n");
    tt=((double)clock())/CLOCKS_PER_SEC;
    qs(a,0,n);
    mm=((double)clock())/CLOCKS_PER_SEC;
    for(int i=0;i<n;i++){
        printf(" %d ",a[i]);
    }
    printf("\n");
    printf("%lf",mm-tt);
    return 0;
}